code/game/atoms.dm

```diff
@@ -1153,3 +1153,10 @@ GLOBAL_LIST_EMPTY(blood_splatter_icons)
 	if(degrees)
 		appearance_flags |= PIXEL_SCALE
 	transform = M
+
+///Setter for the `density` variable to append behavior related to its changing.
+/atom/proc/set_density(new_value)
+	if(density == new_value)
```

Comment:
Spess

Reply:
i am leaving this one, but disregard it its more an eternal reminder for me to get coffee after a nap and before reviewing.